app/services/broadcasts/welcome_notification/generator.rb

```diff
@@ -99,11 +99,14 @@ def commented_on_welcome_thread?
         Comment.where(commentable: welcome_thread, user: user).any?
       end
 
-      def authenticated_with_all_providers?
+      def ga_providers
         # ga_providers refers to Generally Available (not in beta)
-        ga_providers = Authentication::Providers.enabled.reject { |sym| sym == :apple }
-        enabled_providers = identities.pluck(:provider).map(&:to_sym)
-        (ga_providers - enabled_providers).empty?
+        @ga_providers ||=
+          Authentication::Providers.enabled.reject { |sym| sym == :apple }
```

Comment:
this business rule about GA providers probably shouldn't be defined here. I'm happy to move it anywhere else - Authentication::Providers is probably missing a method (enabled, available, ... "suggestable"?)

Reply:
What if this was the generators concern all along - and the key isn't "except apple" but that the broadcast should be inactive/disabled if it's not expected to send?

Something like

```ruby
      def authentication_broadcasts
        Broadcast.active.where(type_of: "Welcome").where("title like '%_connect'")
      end

      def providers
        # we filter the providers to suggest based on the following two criteria
        # - provider is enabled
        # - an acive provider Broadcast message exists
        # Disabling/deactivating a provider_connect welcome message removes
        # the provider from suggestions, while you could
        @providers ||=
          Authentication::Providers.enabled.select do |provider|
            authentication_broadcasts.exists?(title: "Welcome Notification: #{provider}_connect")
          end
      end

      def authenticated_with_all_providers?
        providers.all? { |sym| identities.exists?(provider: sym) }
      end

      def find_auth_broadcast
        missing_identities = providers.filter_map do |provider|
          identities.exists?(provider: provider) ? nil : "#{provider}_connect"
        end

        authentication_broadcasts.find_by!(title: "Welcome Notification: #{missing_identities.sample}")
      end
```